iosApp/iosApp/ui/textFields.swift

```diff
@@ -0,0 +1,93 @@
+import SwiftUI
+
+struct EditText: View {
+    @Binding var input: String
+    var topLabel: String?
+    var errorMessage: String?
+    var shadowRadius: CGFloat = 10
+    var isSecure: Bool = false
+    var body: some View {
+        
+        VStack(alignment: .leading) {
+            if let topLabel = topLabel {
+                Text(topLabel)
+            }
+            if (!isSecure) {
+                TextField("Type here", text: $input)
+                    .padding()
+                    .overlay(
+                        RoundedRectangle(cornerRadius: 10)
+                            .stroke(Color.blackPearl, lineWidth: 1)
+                    )
+                if let errorMessage = errorMessage {
+                    HStack {
+                        Image(systemName: "exclamationmark.circle.fill")
+                            .foregroundColor(.red)
+                        Text(errorMessage)
+                            .foregroundColor(.red)
+                    }
+                }
+            } else {
+                SecureField("Type here", text: $input)
+                    .padding()
+                
+                    .overlay(
+                        RoundedRectangle(cornerRadius: 10)
+                            .stroke(Color.black, lineWidth: 1)
+                    )
+                if let errorMessage = errorMessage {
+                    HStack {
+                        Image(systemName: "exclamationmark.circle.fill")
+                            .foregroundColor(.red)
+                        Text(errorMessage)
+                            .foregroundColor(.red)
+                    }
+                }
+            }
+        }
+    }
+}
+
+
+
+struct PasswordTextField: View {
+    @Binding var password: String
+    var isValid: Bool
+    @Binding var isPasswordVisible: Bool
+    
+    var body: some View {
+        EditText(input: $password, topLabel: "Password", isSecure: !isPasswordVisible)
+            .overlay(
+                Button(action: {
+                    isPasswordVisible.toggle()
+                }) {
+                    Image(systemName: isPasswordVisible ? "eye.slash.fill" : "eye.fill")
+                        .foregroundColor(.blackPearl)
+                        .padding(.top, 30)
+                    
+                }
+                    .padding(.trailing, 8)
```

Comment:
I find it very confusing on what these paddings etc are, is there a way to improve this readability?

Reply:
Is my latest commit better in your opinion? The only downside is that you will always need to specify each "padding direction", so you'll always need to pass in 4 parameters